GentApp/ViewModels/BranchViewModel.cs

```diff
@@ -88,29 +85,30 @@ public class BranchViewModel : ViewModelBase {
 			}
 		}
 
-		public void AddPromotion(Promotion promotion) {
+		public async void AddPromotion(Promotion promotion) {
 			Promotions.Add(promotion);
-
-			SaveBranchCommand.Execute("Promotion");
-			RaisePropertyChanged(nameof(Promotions));
+			await SaveBranch();
+			_navigationService.NavigateTo(nameof(BranchPromotionsPage));
 		}
 
-		public void EditPromotion(string title, string description, DateTime startdate, DateTime enddate) {
+		public async void EditPromotion(string title, string description, DateTime startdate, DateTime enddate) {
 			MySelectedPromotion.Title = title;
 			MySelectedPromotion.Description = description;
 			MySelectedPromotion.StartDate = startdate;
 			MySelectedPromotion.EndDate = enddate;
 
-			Promotions[Promotions.FindIndex(p => p.Id.Equals(MySelectedPromotion.Id))] = MySelectedPromotion;
+			CompanyViewModel.SelectedBranch.Promotions[Promotions.FindIndex(p => p.Id.Equals(MySelectedPromotion.Id))] = MySelectedPromotion;
 
-			SaveBranchCommand.Execute("Promotion");
+			await _promotionService.Update(MySelectedPromotion);
 			RaisePropertyChanged(nameof(Promotions));
+			_navigationService.NavigateTo(nameof(BranchPromotionsPage));
 		}
 
-		public void DeletePromotion() {
-			Promotions.RemoveAt(Promotions.FindIndex(p => p.Id.Equals(MySelectedPromotion.Id)));
-			SaveBranchCommand.Execute("Branch");
+		public async void DeletePromotion() {
+			CompanyViewModel.SelectedBranch.Promotions.RemoveAt(Promotions.FindIndex(p => p.Id.Equals(MySelectedPromotion.Id)));
+			await _promotionService.Delete(MySelectedPromotion);
```

Comment:
same

Reply:
Keb alles moeten aanpassen zodawe enkele requests gebruiken. Tis redelijk ingewikkeld om een parent object up te daten en dan alle children ook up te daten in 1 keer. Ik dacht dat da vanzelf ging maar blijkbaar niet :/